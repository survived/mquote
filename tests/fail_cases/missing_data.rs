extern crate mquote;

use mquote::mquote;

fn main() {
    let _ = mquote!(#{if} 123 #{endif});
    let _ = mquote!(#{if false} 1 #{elif true} 2 #{else} 3 #{endif});
    let _ = mquote!(#{for} repeative #{endfor});
    let _ = mquote!(#{match}#{of Some(x)} #{x} #{endmatch});
    let _ = mquote!(#{match Some("1")} #{of} hi #{endmatch});
}
