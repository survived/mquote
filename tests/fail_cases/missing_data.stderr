error: Missing condition in #{if .. }
 --> $DIR/missing_data.rs:6:23
  |
6 |     let _ = mquote!(#{if} 123 #{endif});
  |                       ^^

error: Missing var and expression in #{for var in expression}
 --> $DIR/missing_data.rs:8:23
  |
8 |     let _ = mquote!(#{for} repeative #{endfor});
  |                       ^^^

error: Missing matching expression in #{match .. }
 --> $DIR/missing_data.rs:9:23
  |
9 |     let _ = mquote!(#{match}#{of Some(x)} #{x} #{endmatch});
  |                       ^^^^^

error: Missing pattern in #{of .. }
  --> $DIR/missing_data.rs:10:42
   |
10 |     let _ = mquote!(#{match Some("1")} #{of} hi #{endmatch});
   |                                          ^^
